api/comments/views.py

```diff
@@ -0,0 +1,140 @@
+from rest_framework.exceptions import NotFound, ValidationError, PermissionDenied
+from api.comments.serializers import CommentSerializer, CommentDetailSerializer, CommentReportsSerializer, CommentReportDetailSerializer, CommentReport
+from api.base.exceptions import Gone
+from api.base import permissions as base_permissions
+from api.nodes.permissions import (
+    ContributorOrPublicForComments
+)
+from framework.auth.oauth_scopes import CoreScopes
+from website.project.model import Comment
+
+
+class CommentMixin(object):
+    """Mixin with convenience methods for retrieving the current comment  based on the
+    current URL. By default, fetches the comment based on the comment_id kwarg.
+    """
+
+    serializer_class = CommentSerializer
+    comment_lookup_url_kwarg = 'comment_id'
+
+    def get_comment(self, check_permissions=True):
+        pk = self.kwargs[self.comment_lookup_url_kwarg]
+        query = Q('_id', 'eq', pk)
+        try:
+            comment = Comment.find_one(query)
+        except NoResultsFound:
+            raise NotFound
+
+        if check_permissions:
+            # May raise a permission denied
+            self.check_object_permissions(self.request, comment)
+        return comment
+
+
+class CommentRepliesList(generics.ListCreateAPIView, CommentMixin):
+    """Replies to a comment.
+
+    By default, a GET will return both deleted and not deleted replies. Comment replies may be
+    filtered by their `deleted` field.
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        ContributorOrPublicForComments,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_COMMENTS_READ]
+    required_write_scopes = [CoreScopes.NODE_COMMENTS_WRITE]
+
+    serializer_class = CommentSerializer
+
+    def get_queryset(self):
+        return Comment.find(Q('target', 'eq', self.get_comment()))
+
+    # overrides ListCreateAPIView
+    def perform_create(self, serializer):
+        self.check_object_permissions(self.request, self.get_comment())
+        serializer.save()
+
+
+class CommentDetail(generics.RetrieveUpdateAPIView, CommentMixin):
+    """Details about a specific comment."""
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        ContributorOrPublicForComments,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_COMMENTS_READ]
+    required_write_scopes = [CoreScopes.NODE_COMMENTS_WRITE]
+
+    serializer_class = CommentDetailSerializer
+
+    # overrides RetrieveAPIView
+    def get_object(self):
+        return self.get_comment()
+
+
+class CommentReportsList(generics.ListCreateAPIView, CommentMixin):
+    """List of reports made for a comment."""
+    permission_classes = (
+        drf_permissions.IsAuthenticated,
+        ContributorOrPublicForComments,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_COMMENTS_READ]
```

Comment:
By using the same Core Scope name for all aspects of comments, all views will be accessible to the outside world equally.

I assume that "comment reports" refer to people flagging a comment (spam etc). What is the use case for making that information available to the outside world? 

(Even if a third-party user wants to see the text of comments, why do they need to see admin data like abuse reports?)